File: frontend/src/build-info.ts
// Vite-injected build metadata (see vite.config.ts `define`).
declare const __APP_VERSION__: string
declare const __GIT_COMMIT__: string
declare const __BUILD_TIME__: string

export interface BuildInfo {
  version: string
  commit: string
  time: string
}

export function useBuildInfo(): BuildInfo {
  return {
    version: __APP_VERSION__,
    commit: __GIT_COMMIT__,
    time: __BUILD_TIME__,
  }
}

File: backend/internal/buildinfo/buildinfo.go
// Package buildinfo carries version metadata injected at build time via
// -ldflags. It has no dependencies so any package can import it.
package buildinfo

// These are overridden at build time:
//
//	-ldflags "-X github.com/oglimmer/coding-agent/backend/internal/buildinfo.Version=1.2.3 ..."
var (
	Version = "dev"
	Commit  = "none"
	Time    = "unknown"
)
